src/Purestone/Game.hs

```diff
@@ -4,19 +4,69 @@
 -- This source code is licensed under the MIT licence found in the           --
 -- LICENSE file in the root directory of this source tree.                   --
 -------------------------------------------------------------------------------
-module Purestone.Game ( setupGame, processMove ) where
+module Purestone.Game ( setupGame, processMove, draw, giveCard, resetPoints,
+        increasePoints, changeTurn ) where
 
 import Purestone.Board
 import Purestone.Card
 import Purestone.Deck
+import Purestone.Player
 
--- | `setupGame` takes the 2 players decks (in a list) and creates the initial 
+import System.Random.Shuffle (shuffleM)
+import System.Random (randomRIO)
+
+-- | `setupGame` takes the 2 players decks (in a list) and creates the initial
 --   board and determines the starting player
-setupGame :: [Deck] -> (Board, Int)
-setupGame = undefined
+setupGame :: (Deck, Deck) -> IO Board
```

Comment:
```suggestion
setupGame :: MonadIO m -> (Deck, Deck) -> m Board
```
I *think* this and the removal of `liftIO` below is the last change we need to merge this

Reply:
Changing just that doesn't work because I don't think MonadIO doesn't implement MonadRandom. I'll be quite busy this week so if anyone else wants to do it, go ahead

Reply:
Ah sorry, didn't realise that, I'll have a look at it later if you're busy